integration-tests/tests/src/tests/migrations.ts

```diff
@@ -0,0 +1,521 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import Realm from "realm";
+import { expect } from "chai";
+import { DogSchema, IDog, IPerson, PersonSchema } from "../schemas/person-and-dogs";
+
+const TestSchema = {
+  name: "Test",
+  properties: {
+    prop0: "string",
```

Comment:
At least in tests, I wouldn't want to make a shared schema as this creates coupling between test suites.  I would like to avoid this, as when one needs to expand on the shared schema, it effects unrelated tests and could lead to unintended failures.

I do agree though, that the property names could be improved.

Reply:
I am changing `prop0 => name, prop1 => age, prop2 => surname`